### PR TITLE
todoist-electron: use electron 24

### PR DESCRIPTION
**Pull request by kalbasit**
###### Description of changes

Todoist electron is still using electron 21 which is EOL. However, it works just fine with the latest electron 24.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -19401,7 +19401,9 @@ with pkgs;
 
   todoist = callPackage ../applications/misc/todoist { };
 
-  todoist-electron = callPackage ../applications/misc/todoist-electron { };
+  todoist-electron = callPackage ../applications/misc/todoist-electron {
+    electron = electron_24;
```
Comment (kalbasit) [inline pkgs/top-level/all-packages.nix:19405]: I'm wondering, should this be unpinned and let it use whatever electron defaults to?

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -19401,7 +19401,9 @@ with pkgs;
...
+  };
```
Comment (kalbasit) [inline pkgs/top-level/all-packages.nix:19406]:
```suggestion
  todoist-electron = callPackage ../applications/misc/todoist-electron { };
```

I'm wondering, should this be unpinned and let it use whatever electron defaults to?

**Comment by kylesferrazza** [comment]
@kalbasit have you tried logging in?

https://github.com/NixOS/nixpkgs/issues/231111#issuecomment-1555904249 points out login breaks when moving from electron 21 to 22.

**Comment by kalbasit** [comment]
> @kalbasit have you tried logging in to electron-24-built-Todoist?
> 
> [#231111 (comment)](https://github.com/NixOS/nixpkgs/issues/231111#issuecomment-1555904249) points out login breaks when moving from electron 21 to 22.

Good catch, I haven't. I'll try right now and report back.

**Comment by kalbasit** [comment]
hmmm it's odd. When logging in, it opens a new window to login but never logs itself back in the main window. I guess we'll have to wait for upstream to fix this.

**Comment by kylesferrazza** [comment]
Superseded by #237663

**Pull request closed without merging**